Plots/Contours/NCL_minmax_3.py

```diff
@@ -0,0 +1,291 @@
+"""
+NCL_minmax_3.py
+===============
+This script illustrates the following concepts:
+   - Calculating the local minima/maxima of your data
+   - Adding text strings at local minima/maxima locations
+   - Generating dummy data using "generate_2d_array"
+   - Setting the background color for a text box
+   - Turning on the perimeter of a text box
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/minmax_3.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/minmax_3_lg.png
+"""
+
+###############################################################################
+# Import packages:
+
+import numpy as np
+import xarray as xr
+import matplotlib.pyplot as plt
+
+from geocat.viz import util as gvutil
+from geocat.viz import cmaps as gvcmaps
+
+################################################################
+# Definition of generate_2d_array and helper functions from https://github.com/NCAR/pyngl/blob/develop/src/ngl/__init__.py
+
+#  Globals for random number generator for generat_2d_array
+dfran_iseq = 0
+dfran_rseq = [.749, .973, .666, .804, .081, .483, .919, .903,   \
+              .951, .960, .039, .269, .270, .756, .222, .478,   \
+              .621, .063, .550, .798, .027, .569, .149, .697,   \
+              .451, .738, .508, .041, .266, .249, .019, .191,   \
+              .266, .625, .492, .940, .508, .406, .972, .311,   \
+              .757, .378, .299, .536, .619, .844, .342, .295,   \
+              .447, .499, .688, .193, .225, .520, .954, .749,   \
+              .997, .693, .217, .273, .961, .948, .902, .104,   \
+              .495, .257, .524, .100, .492, .347, .981, .019,   \
+              .225, .806, .678, .710, .235, .600, .994, .758,   \
+              .682, .373, .009, .469, .203, .730, .588, .603,   \
+              .213, .495, .884, .032, .185, .127, .010, .180,   \
+              .689, .354, .372, .429                            \
+             ]
+
+
+#  Random number generator for generate_2d_array.
+def _dfran():
+    global dfran_iseq
+    global dfran_rseq
+    dfran_iseq = dfran_iseq % 100
+    r = dfran_rseq[dfran_iseq]
+    dfran_iseq = dfran_iseq + 1
+    return r
+
+def generate_2d_array(dims, num_low, num_high, minv, maxv, seed=0, \
+                      highs_at=None, lows_at=None):
+    """Generates a "nice" 2D array of pseudo random data, especially for use in
+    2D graphics.
+
+    dims -- a list (or array) containing the dimensions of the
```

Comment:
I think these should follow the formatting practices we've been using

Reply:
Updated.